src/components/TradeAssetSearch/components/SearchTermAssetList.tsx

```diff
@@ -50,11 +52,19 @@ export const SearchTermAssetList = ({
     [activeChainId, walletConnectedChainIds],
   )
   const walletSupportedEvmChainIds = useMemo(() => chainIds.filter(isEvmChainId), [chainIds])
-  const customTokenSupportedChainIds = useMemo(
-    () =>
-      walletSupportedEvmChainIds.filter(chainId => ALCHEMY_SUPPORTED_CHAIN_IDS.includes(chainId)),
-    [walletSupportedEvmChainIds],
-  )
+  const customTokenSupportedChainIds = useMemo(() => {
+    // If it's a Solana address?
+    if (isValidSolanaAddress(searchString)) {
+      return activeChainId === 'All' || activeChainId === solanaChainId ? [solanaChainId] : []
```

Comment:
nitpick: can we comment out why we need a special case for Solana i.e why Solana isn't present in `ALCHEMY_SUPPORTED_CHAIN_IDS` ? 🙏🏽

Reply:
Yes indeed! https://github.com/shapeshift/web/pull/8619/commits/e50919b753129154f3f1756997551a248ab473e9